internal/apis/certmanager/validation/certificate_test.go

```diff
@@ -944,6 +944,18 @@ func Test_validateLiteralSubject(t *testing.T) {
 			},
 			a: someAdmissionRequest,
 		},
+		"valid with a `literalSubject` containing CN with special characters, multiple DC and well-known rfc4514 and rfc5280 RDN OIDs": {
+			featureEnabled: true,
+			cfg: &internalcmapi.Certificate{
+				Spec: internalcmapi.CertificateSpec{
+					Subject:        &internalcmapi.X509Subject{SerialNumber: "1"},
+					LiteralSubject: "CN=James \\\"Jim\\\" Smith\\, III,DC=dc,DC=net,UID=jamessmith,STREET=La Rambla,L=Barcelona,C=Spain,O=Acme,OU=IT,OU=Admins",
+					SecretName:     "abc",
+					IssuerRef:      validIssuerRef,
+				},
+			},
+			a: someAdmissionRequest,
+		},
```

Comment:
suggestion: This test confirms that we don't error out or produce warnings when we add a DC or UID in the subject, and that's great. It doesn't check that we actually respect the literalSubject when we create a CSR though, as far as I can tell.

We could maybe use some e2e tests of this functionality, similar to what we have for additional output formats [here](https://github.com/cert-manager/cert-manager/blob/8af2d64f3b6374735b0204f173ceb78d066a30ff/test/e2e/suite/certificates/additionaloutputformats.go).

Do you think you could create a really simple e2e test similar to [those for additional output formats](https://github.com/cert-manager/cert-manager/blob/8af2d64f3b6374735b0204f173ceb78d066a30ff/test/e2e/suite/certificates/additionaloutputformats.go), creating a certificate with the literalsubject from this test and confirming that the CSR subject matches what's expected?

I realise that's not a trivial bit of work, but I think it'd let us be a lot more confident about this feature.

Reply:
Ok PTAL with the latest changes in `test/e2e/suite/certificates`